iOS/Flows/Conversation/Conversation List/ConversationListViewController+Extensions.swift

```diff
@@ -52,34 +52,34 @@ extension ConversationListViewController {
                                                  collectionView: self.collectionView)
                 }
         }.store(in: &self.cancellables)
-//
-//        self.conversationController?.typingUsersPublisher.mainSink { [unowned self] users in
-//            let nonMeUsers = users.filter { user in
-//                return user.userObjectID != User.current()?.objectId
-//            }
-//            self.messageInputAccessoryView.updateTypingActivity(with: nonMeUsers)
-//        }.store(in: &self.cancellables)
 
         self.collectionView.publisher(for: \.contentOffset).mainSink { [unowned self] _ in
             guard self.collectionView.isTracking else { return }
             self.collectionView.visibleCells.forEach { cell in
-                if let messageCell = cell as? ConversationMessageCell {
+                if let messageCell = cell as? ConversationMessagesCell {
                     messageCell.handle(isCentered: false)
                 }
             }
         }.store(in: &self.cancellables)
 
-//        self.messageInputAccessoryView.textView.$inputText.mainSink { [unowned self] _ in
-//            guard let enabled = self.conversationController?.areTypingEventsEnabled, enabled else { return }
-//            self.conversationController?.sendKeystrokeEvent(completion: nil)
-//        }.store(in: &self.cancellables)
+        self.messageInputAccessoryView.textView.$inputText.mainSink { [unowned self] text in
+            guard let conversationController = self.conversationController else { return }
+
+            guard conversationController.areTypingEventsEnabled else { return }
+
+            if text.isEmpty {
+                conversationController.sendStopTypingEvent()
+            } else {
+                conversationController.sendKeystrokeEvent()
+            }
```

Comment:
Restoring the typing indicator events. The actual indicator isn't showing up yet.